### PR TITLE
T004 on off effort

### PR DESCRIPTION
**Pull request by Fred1511**

### CODE REVIEW

File: sammo.py [hunk 1]
```diff
@@ -4,17 +4,44 @@
...
-        self.actionSession = SammoActionSession(self.iface.mainWindow(), self.toolBar)
+        self._toolBar: QToolBar = self.iface.addToolBar("Sammo ToolBar")
+        self._session = SammoSession()
+
+        self.actionSession = SammoActionSession(iface.mainWindow(), self._toolBar)
+        self.actionSession.createSignal.connect(self.onCreateSession)
+
+        self.actionOnOffSession = SammoActionOnOffEffort(iface.mainWindow(), self._toolBar)
+        self.actionOnOffSession.onStartEffortSignal.connect(self.onStartEffort)
+        self.actionOnOffSession.onStopEffortSignal.connect(self.onStopEffort)
```
Comment (pblottiere) [inline sammo.py:24]: The same signal can be used with a boolean parameter.

File: sammo.py [hunk 2]
```diff
@@ -4,17 +4,44 @@
...
+        self.actionOnOffSession.onAddFeatureToEnvironmentTableSignal.connect(self.onAddFeatureToEnvironmentTableSignal)
 
     def initGui(self):
-        self.actionSession.initGui()
+        pass
```
Comment (pblottiere) [inline sammo.py:28]: We can remove `initGui`

File: src/core/session.py [hunk 3]
```diff
@@ -3,24 +3,85 @@
...
+
+    def addNewFeatureToEnvironmentTable(self, feat):
+        self._environmentTable.addFeature(feat)
+        self._environmentTable.commitChanges()
+
+    def _getReadyToAddNewFeature(self, table: QgsVectorLayer):
+        feat = QgsVectorLayerUtils.createFeature(table)
+        table.startEditing()
+
+        return [feat, table]
```
Comment (pblottiere) [inline src/core/session.py:87]: The list is unnecessary, you can just write `return feat, table`. Then you retrieve the corresponding values with `feat, table = self._getReadyToAddNewFeature(...)`

File: src/gui/onOffEffort.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    def unload(self):
+        del self.button
+
+    def run(self):
+        if self.button.isChecked():
+            self.onStartEffortSignal.emit()
+        else:
+            self.onStopEffortSignal.emit()
+
+    def OpenFeatureForm(self, iface, table: QgsVectorLayer, feat: QgsFeature):
```
Comment (pblottiere) [inline src/gui/onOffEffort.py:42]: camelCase

File: src/core/session.py (hunk 3 above)
Comment (pblottiere) [inline src/core/session.py:87]: The list is unnecessary, you can just write `return feat, table`. Then you retrieve the corresponding values with `feat, table = self._getReadyToAddNewFeature(...)`

File: src/gui/onOffEffort.py (hunk 4 above)
Comment (pblottiere) [inline src/gui/onOffEffort.py:42]: camelCase

File: src/gui/onOffEffort.py [hunk 5]
```diff
@@ -0,0 +1,44 @@
+# coding: utf8
```
Comment (pblottiere) [inline src/gui/onOffEffort.py:1]: The file should be renamed in `on_off_effort.py` (or maybe just `effort.py`?)

File: sammo.py (hunk 2 above)
Comment (Fred1511) [reply sammo.py:28]: I tried but got a QGIS error. It seems to be a mandatory method

Comment (pblottiere) [reply sammo.py:27]: Ok

**Comment by pblottiere** [comment]
@Fred1511 you just have to fix the CI (flake8/black) and we're good to merge then :)

**Comment by pblottiere** [comment]
Thanks @Fred1511 :+1:

**Pull request merged**